src/Module/Login.php

```diff
@@ -347,4 +378,56 @@ public static function form($return_path = null, $register = false, $hiddens = [
 
 		return $o;
 	}
+
+	/**
+	 * Get the URL to the register page and add OpenID parameters to it
+	 */
+	private static function getRegisterURL()
+	{
+		if (empty(Session::get('openid_identity'))) {
+			return 'register';
+		}
+
+		$args = '';
+		$attr = Session::get('openid_attributes') ?? [];
+
+		if (is_array($attr) && count($attr)) {
+			foreach ($attr as $k => $v) {
+				if ($k === 'namePerson/friendly') {
+					$nick = Strings::escapeTags(trim($v));
+				}
+				if ($k === 'namePerson/first') {
+					$first = Strings::escapeTags(trim($v));
+				}
+				if ($k === 'namePerson') {
+					$args .= '&username=' . urlencode(Strings::escapeTags(trim($v)));
+				}
+				if ($k === 'contact/email') {
+					$args .= '&email=' . urlencode(Strings::escapeTags(trim($v)));
+				}
+				if ($k === 'media/image/aspect11') {
+					$photosq = bin2hex(trim($v));
+				}
+				if ($k === 'media/image/default') {
+					$photo = bin2hex(trim($v));
+				}
+			}
+		}
+
+		if (!empty($nick)) {
+			$args .= '&nickname=' . urlencode($nick);
+		} elseif (!empty($first)) {
+			$args .= '&nickname=' . urlencode($first);
+		}
+
+		if (!empty($photosq)) {
+			$args .= '&photo=' . urlencode($photosq);
+		} elseif (!empty($photo)) {
+			$args .= '&photo=' . urlencode($photo);
+		}
+
+		$args .= '&openid_url=' . urlencode(Strings::escapeTags(trim(Session::get('openid_identity'))));
```

Comment:
Please use `http_build_query()` instead, you don't have to care for `urlencode()` calls or adding `&` and can use a simple associative array with query string keys and values.

Reply:
Yeah, this is some copied code. 
BTW: I'm always searching for that function - but I'm always forgetting the name :-)

Reply:
Done